subsys/bluetooth/audio/shell/bap.c

```diff
@@ -2562,47 +2613,40 @@ static int cmd_send(const struct shell *sh, size_t argc, char *argv[])
 }
 
 #if defined(CONFIG_LIBLC3)
-static bool stream_start_sine_verify(const struct bt_bap_stream *bap_stream)
+static bool stream_start_sine_verify(const struct shell_stream *sh_stream)
 {
+	const struct bt_bap_stream *bap_stream;
 	struct bt_bap_ep_info info;
 	int err;
 
-	if (bap_stream == NULL || bap_stream->qos == NULL) {
+	if (sh_stream == NULL) {
 		return false;
 	}
 
-	err = bt_bap_ep_get_info(bap_stream->ep, &info);
-	if (err != 0) {
+	bap_stream = &sh_stream->stream.bap_stream;
```

Comment:
We are guaranteed that bap_stream is non-NULL since the unicast_stream gets initialised in the function cmd_init and so we do not need to check for NULL.
Is that correct?

Reply:
No: We are guaranteed that `bap_stream` is non NULL because it's the address of a field :) (Notice the `&`)